Tests/SwiftDocCTests/Semantics/SymbolTests.swift

```diff
@@ -1111,7 +1112,120 @@ class SymbolTests: XCTestCase {
         let _ = DocumentationNode.contentFrom(documentedSymbol: symbol, documentationExtension: nil, engine: engine)
         XCTAssertEqual(engine.problems.count, 0)
     }
-    
+
+    func testAddingConstraintsToSymbol() throws {
+        let (withoutArticle, _) = try makeDocumentationNodeSymbol(
+            docComment: """
+                A cool API to call.
+
+                - Parameters:
+                  - name: A parameter
+                - Returns: Return value
+                """,
+            articleContent: nil
+        )
+
+        // This symbol starts with 3 generic constraints. See:
+
+        // jq < "Tests/SwiftDocCTests/Test Bundles/TestBundle.docc/mykit-iOS.symbols.json" '.symbols[] | select(.identifier.precise == "s:5MyKit0A5ClassC10myFunctionyyF") | .swiftExtension'
+        // {
+        //   "extendedModule": "MyKit",
+        //   "constraints": [
+        //     {
+        //       "kind": "sameType",
+        //       "lhs": "Label",
+        //       "rhs": "Text"
+        //     },
+        //     {
+        //       "kind": "superclass",
+        //       "lhs": "Observer",
+        //       "rhs": "NSObject"
+        //     },
+        //     {
+        //       "kind": "conformance",
+        //       "lhs": "S",
+        //       "rhs": "StringProtocol"
+        //     }
+        //   ]
+        // }
+        XCTAssertEqual("MyKit", withoutArticle.extendedModuleVariants[DocumentationDataVariantsTrait.swift])
+        let constraints = try XCTUnwrap(withoutArticle.constraints)
+        XCTAssertEqual(3, constraints.count)
+        var constraint = constraints[0]
+        XCTAssertEqual(SymbolGraph.Symbol.Swift.GenericConstraint.Kind.sameType, constraint.kind)
```

Comment:
FYI: The compiler should be able to infer the types here allowing these assertions to be shorter and more readable
```suggestion
        XCTAssertEqual(.sameType, constraint.kind)
```

Reply:
That's great - it must be that since I wrote the expected value first, the compiler/Xcode showed me an error until I finished typing the line. I wonder if it makes more sense to write the actual value first inside of XCTAssertEqual for this reason...